app/models/katello/glue/candlepin/pool.rb

```diff
@@ -126,7 +127,11 @@ def import_data
           pool_attributes[:unmapped_guest] = true
         end
 
-        pool_attributes[:virt_who] = pool_attributes['virt_limit'] != "0" && !pool_attributes['virt_limit'].nil? && subscription.try(:redhat?)
+        if subscription.try(:redhat?)
+          pool_attributes[:virt_who] = pool_attributes['virt_limit'] != "0" && !pool_attributes['virt_limit'].nil?
```

Comment:
I think maybe he's suggesting an additional check or to simplify it down to one.